cupyx/scipy/interpolate/_bspline.py

```diff
@@ -0,0 +1,912 @@
+
+import operator
+from itertools import product
+
+import cupy
+from cupy._core import internal
+from cupy._core._scalar import get_typename
+
+from cupyx.scipy.sparse import csr_matrix
+
+import numpy as np
+
+TYPES = ['float', 'double', 'complex<float>', 'complex<double>']
+INT_TYPES = ['int', 'long long']
+
+INTERVAL_KERNEL = r'''
+#include <cupy/complex.cuh>
+
+template<typename T>
+__global__ void find_interval(
+        const T* t, const T* x, long long* out,
```

Comment:
So, it will get templated on the TYPES above, correct?  `x` and `t` are evaluation points and knots, they are certainly real-valued only. IIRC, the scipy analog upcasts floats to doubles even.